packages/editor/src/browser/editor-command.ts

```diff
@@ -165,6 +185,76 @@ export class EditorCommandContribution implements CommandContribution {
             editor.setLanguage(selected.id);
         }
     }
+
+    protected canConfigureEncoding(): boolean {
+        const widget = this.editorManager.currentEditor;
+        const editor = widget && widget.editor;
+        return !!editor;
+    }
+    protected async configureEncoding(): Promise<void> {
+        const widget = this.editorManager.currentEditor;
+        const editor = widget && widget.editor;
+        if (!editor) {
+            return;
+        }
+        const reopenWithEncodingPick = { label: 'Reopen with Encoding', value: 'reopen' };
+        const saveWithEncodingPick = { label: 'Save with Encoding', value: 'save' };
+        const actionItems: QuickPickItem<string>[] = [
+            reopenWithEncodingPick,
+            saveWithEncodingPick
+        ];
+        const action = await this.quickPick.show(actionItems, {
+            placeholder: 'Select Action'
+        });
+        if (!action) {
+            return;
+        }
+        const isReopenWithEncoding = (action === reopenWithEncodingPick.value);
+
+        const configuredEncoding = this.preferencesService.get(EditorCommandContribution.DEFAULT_ENCODING_PREFERENCE);
```

Comment:
you can use editor preferences instead to get a statically typed interface

Reply:
Fixing soon...